### DISCUSSION TITLE
How to work with typed props?

### DISCUSSION THREAD

**Discussion by igorjacauna** [histoire-dev/histoire#149] [Q&A]
I use with `Button` component:

On `Button.vue`

```vue 
<script lang="ts" setup>
type ButtonProps = {
  type?: Type;
  size?: Size;
  disabled?: boolean;
}

const props = withDefaults(defineProps<ButtonProps>(), {
  type: 'default',
  size: 'normal',
  disabled: false,
});
</script>
```

On `types.d.ts` have:

```ts
declare type Size = 'normal' | 'large' | 'small';
declare type Type = 'default' | 'primary' | 'danger';
```

But on Controls shows `textarea` that only accepts JSON:

![image](https://user-images.githubusercontent.com/459127/175069652-8f4bf62d-ac24-445f-ae48-41f33984ab59.png)

How correct configure the Controls?

**Comment by victorlmneves** [comment]
@igorjacauna do you want to show the input text instead of the textarea?

**Reply by emersonbottero** [reply]
I would like radio buttons
If we have more than 5 options than a select input .

**Reply by victorlmneves** [reply]
then you have to set a condition to render radio buttons or the select within the story

**Reply by emersonbottero** [reply]
I mean when the type is a join between strings as in the above examples this renders should be the default ones. because I would use them in all my one hundred components.. 😁.
can I change the renderer for all histories in a single place?
